pkg/dom0-ztools/rootfs/etc/init.d/000-kexec

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+#
+# Kexec setup before everything else.
+#
+if cat /proc/cmdline | grep -q crashkernel; then
+    #
+    # Don't load kernel if no `crashkernel=Y@X` cmdline.
+    # This means we are on arch which does not support kdump
+    # (e.g. riscv64) or xen (see pkg/grub/rootfs.cfg)
+    #
+    /usr/sbin/kexec -p --reuse-cmdline --command-line="init 1 irqpoll maxcpus=1 reset_devices" /boot/kernel
+elif test -f /proc/vmcore; then
+    #
+    # We are in dump capture kernel, nice. Remount /etc as rw to
+    # stop further containerd execution.
+    #
+    mkdir -p /tmp/ofs/rootfs.rw
+    mkdir -p /tmp/ofs/upper
+    mkdir -p /tmp/ofs/work
+    mount -t overlay overlay -o lowerdir=/,upperdir=/tmp/ofs/upper,workdir=/tmp/ofs/work /tmp/ofs/rootfs.rw
+    mount --bind /tmp/ofs/rootfs.rw/etc /etc
+
+    #
+    # To stop further containerd execution just make the init.d
+    # script as NOP. Ugly, but deleting this script results in
+    # stat errors output, so keep the whole log clean.
+    #
+    rm /etc/init.d/020-containerd
+    echo '#!/bin/sh' > /etc/init.d/020-containerd
+    chmod +x /etc/init.d/020-containerd
```

Comment:
Can I, please, see exactly what kind of errors are you seeing? Otherwise it is difficult to suggest alternatives (but I'm pretty sure we can get rid of this block)

Reply:
Hold on, will take a while (need rebuild etc). As I remember errors come from initd which does "readdir", starts iterating over files and then suddenly a file disappears, so "stat" syscall fails.

Reply:
Nothing critical, but this error output really annoys.

Reply:
So here it is the error which you see if you just 'rm /etc/init.d/020-containerd':

`2022/09/12 11:55:30 Cannot stat /etc/init.d/020-containerd: stat /etc/init.d/020-containerd: no such file or directory`

I assume this complain comes from the `init`, which caches all the files in a list before processing them (which is obvious) and then suddenly finds out the file is missing. 

This is what you wanted to see, @rvs ?

Reply:
As we discussed, I think it is sufficient and less clunky to make the if test -f /proc/vmcore; part of this script into an onboot container which runs after storage-init, and if /proc/vmcore, after it has saved things away in /persist, it reboots to get back to running the normal system.

That avoids trying to disable linuxkit from running service containers etc.